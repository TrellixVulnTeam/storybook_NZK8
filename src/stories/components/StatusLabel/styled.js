// @flow

import styled from 'styled-components';
import { colors, statusColors } from 'utils/StyleGuide';

/**
 * Determine label icon size based on size prop
 * @param {string} size
 * @return {string}
 */
function getSize(size) {
  switch (size) {
    case 'sm':
      return { width: 8, height: 8, borderWidth: 1 };
    default:
      return { width: 14, height: 14, borderWidth: 2 };
  }
}

export const StatusLabelWrapper = styled.span`
  display: flex;
  align-items: center;
`;

export const StatusIcon = styled.span`
  position: relative;
  width: ${p => getSize(p.size).width}px;
  height: ${p => getSize(p.size).height}px;
  border-radius: 50%;
  background-color: ${p => (p.status === 'OPEN' ? colors.transparent : statusColors[p.status])};
  border-width: ${p => p.status === 'OPEN' && `${getSize(p.size).borderWidth}px`};
  border-style: ${p => p.status === 'OPEN' && 'solid'};
  border-color: ${p => p.status === 'OPEN' && statusColors.border[p.status]};
  margin-right: ${p => JSON.parse(p.label) && '5px'};
  transform: ${p => p.offset && p.index !== 0 && `translateX(${p.offset * p.index * -1}px)`};
  box-shadow: ${p => p.offset && p.index !== 0 && `inset 1px 0 0 ${colors.white}`};
`;

StatusIcon.defaultProps = {
  label: 'true',
};

export const StatusLabel = styled.span`
  display: ${p => (JSON.parse(p.label) ? 'inline' : 'none')};

  @media (max-width: 875px) {
    display: none;
  }
`;

StatusLabel.defaultProps = {
  label: 'true',
};

export const CheckIcon = styled.span`
  &::after {
    content: '';
    display: ${p => (p.size === 'sm' ? 'none' : 'block')};
    position: absolute;
    top: 3px;
    left: 6px;
    width: 2px;
    height: 6px;
    border: solid ${colors.white};
    border-width: 0 1px 1px 0;
    transform: rotate(45deg);
  }
`;
