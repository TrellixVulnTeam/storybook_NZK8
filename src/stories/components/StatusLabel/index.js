// @flow

import React from 'react';
import { StatusLabelWrapper, StatusIcon, StatusLabel, CheckIcon } from './styled';

type Props = {
  status: string,
  label?: string,
  size?: 'sm',
  offset?: number,
  index?: number,
};

export default (props: Props) => {
  const displayStatus = {
    OPEN: 'Open',
    IN_PROGRESS: 'In Progress',
    ON_HOLD: 'On Hold',
    COMPLETE: 'Complete',
  };

  const isComplete = props.status === 'COMPLETE';

  return (
    <StatusLabelWrapper
      status={props.status}
      label={props.label}
      size={props.size}
      offset={props.offset}
      index={props.index}
    >
      <StatusIcon {...props}>{isComplete && <CheckIcon size={props.size} />}</StatusIcon>
      <StatusLabel {...props}>{displayStatus[props.status]}</StatusLabel>
    </StatusLabelWrapper>
  );
};
