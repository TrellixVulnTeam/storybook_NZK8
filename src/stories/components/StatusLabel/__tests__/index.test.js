import React from 'react';
import { mount } from 'enzyme';
import StatusLabel from 'components/StatusLabel/index';

describe('<StatusLabel />', () => {
  const mandatoryProps = {
    status: 'COMPLETE',
  };

  it('should render correctly with mandatory props', () => {
    const sut = mount(<StatusLabel {...mandatoryProps} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly if not complited with optional props', () => {
    const props = {
      ...mandatoryProps,
      status: 'OPEN',
      label: '"label"',
      size: 'sm',
      offset: 10,
      index: 1,
    };
    const sut = mount(<StatusLabel {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly if completed with optional props', () => {
    const props = {
      ...mandatoryProps,
      status: 'COMPLETE',
      label: 'false',
      size: 'sm',
      offset: 10,
      index: 1,
    };
    const sut = mount(<StatusLabel {...props} />);

    expect(sut).toMatchSnapshot();
  });
});
