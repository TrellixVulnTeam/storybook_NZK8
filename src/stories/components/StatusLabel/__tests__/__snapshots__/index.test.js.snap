// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusLabel /> should render correctly if completed with optional props 1`] = `
<_default
  index={1}
  label="false"
  offset={10}
  size="sm"
  status="COMPLETE"
>
  <styled__StatusLabelWrapper
    index={1}
    label="false"
    offset={10}
    size="sm"
    status="COMPLETE"
  >
    <span
      className="styled__StatusLabelWrapper-sc-1pj2h9f-0 gxWZId"
      label="false"
      offset={10}
      size="sm"
    >
      <styled__StatusIcon
        index={1}
        label="false"
        offset={10}
        size="sm"
        status="COMPLETE"
      >
        <span
          className="styled__StatusIcon-sc-1pj2h9f-1 cEubhv"
          label="false"
          offset={10}
          size="sm"
        >
          <styled__CheckIcon
            size="sm"
          >
            <span
              className="styled__CheckIcon-sc-1pj2h9f-3 geKZvV"
              size="sm"
            />
          </styled__CheckIcon>
        </span>
      </styled__StatusIcon>
      <styled__StatusLabel
        index={1}
        label="false"
        offset={10}
        size="sm"
        status="COMPLETE"
      >
        <span
          className="styled__StatusLabel-sc-1pj2h9f-2 fyrmuP"
          label="false"
          offset={10}
          size="sm"
        >
          Complete
        </span>
      </styled__StatusLabel>
    </span>
  </styled__StatusLabelWrapper>
</_default>
`;

exports[`<StatusLabel /> should render correctly if not complited with optional props 1`] = `
<_default
  index={1}
  label="\\"label\\""
  offset={10}
  size="sm"
  status="OPEN"
>
  <styled__StatusLabelWrapper
    index={1}
    label="\\"label\\""
    offset={10}
    size="sm"
    status="OPEN"
  >
    <span
      className="styled__StatusLabelWrapper-sc-1pj2h9f-0 gxWZId"
      label="\\"label\\""
      offset={10}
      size="sm"
    >
      <styled__StatusIcon
        index={1}
        label="\\"label\\""
        offset={10}
        size="sm"
        status="OPEN"
      >
        <span
          className="styled__StatusIcon-sc-1pj2h9f-1 fbADtP"
          label="\\"label\\""
          offset={10}
          size="sm"
        />
      </styled__StatusIcon>
      <styled__StatusLabel
        index={1}
        label="\\"label\\""
        offset={10}
        size="sm"
        status="OPEN"
      >
        <span
          className="styled__StatusLabel-sc-1pj2h9f-2 jGdXWF"
          label="\\"label\\""
          offset={10}
          size="sm"
        >
          Open
        </span>
      </styled__StatusLabel>
    </span>
  </styled__StatusLabelWrapper>
</_default>
`;

exports[`<StatusLabel /> should render correctly with mandatory props 1`] = `
<_default
  status="COMPLETE"
>
  <styled__StatusLabelWrapper
    status="COMPLETE"
  >
    <span
      className="styled__StatusLabelWrapper-sc-1pj2h9f-0 gxWZId"
    >
      <styled__StatusIcon
        label="true"
        status="COMPLETE"
      >
        <span
          className="styled__StatusIcon-sc-1pj2h9f-1 gABfOm"
          label="true"
        >
          <styled__CheckIcon>
            <span
              className="styled__CheckIcon-sc-1pj2h9f-3 ecdigu"
            />
          </styled__CheckIcon>
        </span>
      </styled__StatusIcon>
      <styled__StatusLabel
        label="true"
        status="COMPLETE"
      >
        <span
          className="styled__StatusLabel-sc-1pj2h9f-2 jGdXWF"
          label="true"
        >
          Complete
        </span>
      </styled__StatusLabel>
    </span>
  </styled__StatusLabelWrapper>
</_default>
`;
